### PR TITLE
[Gate Tests] - marking feature tests

### PR DESCRIPTION
**Pull request by cseale**
Removal of the lang feature gate tests whitelist #39059

r? @est31

### CODE REVIEW

**Comment by est31** [comment]
Thanks!

As I said in #39059 we need to wait until #39247 is merged (+ then you'll have to rebase), as it requires the check to be implemented.

**Comment by cseale** [comment]
np

**Comment by est31** [comment]
@cseale the PR got merged.

**Review by est31 — APPROVED** [review]

File: src/tools/tidy/src/features.rs [hunk 1]
```diff
@@ -166,8 +166,7 @@ pub fn check(path: &Path, bad: &mut bool) {
 
     // FIXME get this whitelist empty.
     let whitelist = vec![
-        "abi_ptx", "simd", "macro_reexport",
-        "static_recursion", "quote",
+        "abi_ptx", "simd", "static_recursion", "reflect",
```
Comment (jseyfried) [inline src/tools/tidy/src/features.rs]: Why was `"reflect"` added here?

Comment (est31) [inline src/tools/tidy/src/features.rs]: It was removed by https://github.com/rust-lang/rust/pull/39075, so shouldn't be present.

Comment (cseale) [inline src/tools/tidy/src/features.rs]: cleaned this up

**Review by est31 — APPROVED** [review]

File: src/tools/tidy/src/features.rs [hunk 2]
```diff
@@ -166,8 +166,7 @@ pub fn check(path: &Path, bad: &mut bool) {
 
     // FIXME get this whitelist empty.
     let whitelist = vec![
-        "abi_ptx", "simd", "macro_reexport",
-        "static_recursion", "quote",
+        "abi_ptx", "simd", "static_recursion",
```
Comment (est31) [inline src/tools/tidy/src/features.rs]: You've got a trailing whitespace here, that's why tidy fails.

**Pull request merged**